packages/headless/src/api/analytics/insight-analytics.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  AnalyticsClientSendEventHook,
+  InsightClientProvider,
+  CoveoInsightClient,
+  history,
+} from 'coveo.analytics';
+import {SearchEventRequest} from 'coveo.analytics/dist/definitions/events';
+import {Logger} from 'pino';
+import {
+  buildFacetStateMetadata,
+  getStateNeededForFacetMetadata,
+  SectionNeededForFacetMetadata,
+} from '../../features/facets/facet-set/facet-set-analytics-actions-utils';
+import {getQueryInitialState} from '../../features/query/query-state';
+import {getSearchInitialState} from '../../features/search/search-state';
+import {
+  ConfigurationSection,
+  PipelineSection,
+  QuerySection,
+  SearchHubSection,
+  SearchSection,
+} from '../../state/state-sections';
+import {PreprocessRequest} from '../preprocess-request';
+import {BaseAnalyticsProvider} from './base-analytics';
+
+export type StateNeededByInsightAnalyticsProvider = ConfigurationSection &
+  Partial<
+    SearchHubSection &
+      SearchSection &
+      PipelineSection &
+      QuerySection &
+      SectionNeededForFacetMetadata
+  >;
+
+export class InsightAnalyticsProvider
+  extends BaseAnalyticsProvider
+  implements InsightClientProvider
+{
+  public getSearchUID(): string {
+    return (
+      this.state.search?.searchResponseId ||
+      this.state.search?.response.searchUid ||
+      getSearchInitialState().response.searchUid
+    );
+  }
+  public getPipeline(): string {
+    return (
+      this.state.pipeline || this.state.search?.response.pipeline || 'default'
+    );
+  }
+  public getSearchEventRequestPayload(): Omit<
+    SearchEventRequest,
+    'actionCause' | 'searchQueryUid'
+  > {
+    return {
+      queryText: this.queryText,
+      responseTime: this.responseTime,
+      results: this.mapResultsToAnalyticsDocument(),
+      numberOfResults: this.numberOfResults,
+    };
+  }
+  constructor(private state: StateNeededByInsightAnalyticsProvider) {
+    super(state);
+  }
+
+  public getFacetState() {
+    return buildFacetStateMetadata(getStateNeededForFacetMetadata(this.state));
+  }
+
+  private get queryText() {
+    return this.state.query?.q || getQueryInitialState().q;
+  }
+
+  private get responseTime() {
+    return this.state.search?.duration || getSearchInitialState().duration;
+  }
+
+  private mapResultsToAnalyticsDocument() {
+    return this.state.search?.response.results.map((r) => ({
+      documentUri: r.uri,
+      documentUriHash: r.raw.urihash,
+    }));
```

Comment:
permanenentId? Not 100% of the reference for this as currently it doesn't look like we send the results on search events. But at least for clicks we definitely need the permanentId

Reply:
Okay well I validated and this is what atomic does as well I guess that is okay?

I see no mention of the results in this page: https://docs.coveo.com/en/1502/build-a-search-ui/log-search-events
Or even in the swagger doc

Reply:
It was part of the base PR https://github.com/coveo/ui-kit/pull/2147